oauthproxy.go

```diff
@@ -838,6 +839,27 @@ func (p *OAuthProxy) enrichSessionState(ctx context.Context, s *sessionsapi.Sess
 func (p *OAuthProxy) AuthOnly(rw http.ResponseWriter, req *http.Request) {
 	session, err := p.getAuthenticatedSession(rw, req)
 	if err != nil {
+		if authOnlyRedirectSignIn(req) {
+			if isAjax(req) {
+				// no point redirecting an AJAX request
+				p.errorJSON(rw, http.StatusUnauthorized)
+				return
+			}
+
+			// Orignal request URI is captured and added using `rd` parameter in the redirect
+			// to ensure a redirect to the original URL will still happen after OAuth/Signin process
+			redirect, err := p.getAppRedirect(req)
+			if err != nil {
+				logger.Errorf("Error obtaining redirect: %v", err)
+				p.ErrorPage(rw, http.StatusInternalServerError, "Internal Server Error", err.Error())
```

Comment:
Make a reminder note:

https://github.com/oauth2-proxy/oauth2-proxy/pull/1029 will likely merge before this. When you rebase, you'll need to update to use the new style of error pages.

Reply:
updated